src/zenml/annotators/base_annotator.py

```diff
@@ -0,0 +1,140 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Base class for ZenML annotator stack components."""
+
+from abc import ABC, abstractmethod
+from typing import Any, ClassVar, List, Optional, Tuple
+
+from zenml.enums import StackComponentType
+from zenml.stack import StackComponent
+
+
+class BaseAnnotator(StackComponent, ABC):
+    """Base class for all ZenML annotators.
+
+    Attributes:
+        notebook_only: if the annotator can only be used in a notebook.
+    """
+
+    notebook_only: ClassVar[bool] = False
+
+    # Class configuration
+    TYPE: ClassVar[StackComponentType] = StackComponentType.ANNOTATOR
+    FLAVOR: ClassVar[str]
+
+    @abstractmethod
+    def get_url(self) -> str:
+        """Gets the URL of the annotation interface.
+
+        Returns:
+            The URL of the annotation interface.
+        """
+
+    @abstractmethod
+    def get_annotation_url(self, dataset_name: str) -> str:
+        """Gets the URL of the annotation interface for a specific dataset.
+
+        Args:
+            dataset_name: name of the dataset.
+
+        Returns:
+            The URL of the dataset annotation interface.
+        """
+
+    @abstractmethod
+    def get_datasets(self) -> List[Any]:
+        """Gets the datasets currently available for annotation.
+
+        Returns:
+            The datasets currently available for annotation.
+        """
+
+    @abstractmethod
+    def get_dataset_names(self) -> List[str]:
+        """Gets the names of the datasets currently available for annotation.
+
+        Returns:
+            The names of the datasets currently available for annotation.
+        """
+
+    @abstractmethod
+    def get_dataset_stats(self, dataset_name: str) -> Tuple[int, int]:
```

Comment:
is Tuple[int, int] general enough for the return of dataset stats for other annotation tools not only label studio?

Reply:
Generally you get the number of unlabeled and labeled examples for a particular project. That's been the same across most of the tools I've used.

That said, I have no problem making a breaking change later :)